### PR TITLE
*: let location-labels works if only exist the default rule

### PR DESCRIPTION
**Pull request by nolouch**
Signed-off-by: nolouch <nolouch@gmail.com>

### What problem does this PR solve?

Compatible with 4.0 default behavior，the default config can set with :
```
config set max-replicas xxx
config set location-labels xxx,xxx
```

### Check List

Tests

- Unit test

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -846,21 +846,35 @@ func (s *Server) SetReplicationConfig(cfg config.ReplicationConfig) error {
...
+		CheckInDefaultRule := func() error {
+			// replication config  won't work when placement rule is enabled and exceeds one default rule
+			if !(len(rules) == 1 &&
+				len(rules[0].StartKey) == 0 && len(rules[0].EndKey) == 0 &&
+				rules[0].GroupID == "pd" && rules[0].ID == "default") {
+				return errors.New("cannot update MaxReplicas or LocationLabels when placement rules feature is enabled and not only default rule exists, please update rule instead")
 			}
 			rule = rules[0]
+			if !(rule.Count == int(old.MaxReplicas) && reflect.DeepEqual(rule.LocationLabels, []string(old.LocationLabels))) {
+				fmt.Println(rule.Count, old.MaxReplicas, old.LocationLabels, rule.LocationLabels)
```
Comment (HunDunDM) [inline server/server.go]: remove this line

**Review by HunDunDM — APPROVED** [review]

**Review by lhy1024 — APPROVED** [review]

**Pull request merged**